apps/admin-gui/src/app/shared/components/dialogs/create-attribute-definition-dialog/create-attribute-definition-dialog.component.ts

```diff
@@ -80,44 +80,116 @@ export class CreateAttributeDefinitionDialogComponent implements OnInit {
+      rightsSELF.rights.push(ActionType.WRITE_PUBLIC);
     }
     if (this.writeSelfVo) {
-      list.push(this.generateRights(Role.SELF, ActionType.WRITE_VO));
+      rightsSELF.rights.push(ActionType.WRITE_VO);
     }
+
+    list.push(rightsSELF);
+
+    const rightsVO = {} as AttributeRights;
+    rightsVO.attributeId = this.attDef.id;
+    rightsVO.role = Role.VOADMIN;
+    rightsVO.rights = [];
+
+    if (this.readVo) {
+      rightsVO.rights.push(ActionType.READ);
+    }
+
     if (this.writeVo) {
-      list.push(this.generateRights(Role.VOADMIN, ActionType.WRITE));
+      rightsVO.rights.push(ActionType.WRITE);
     }
+
+    list.push(rightsVO);
+
+    const rightsGROUP = {} as AttributeRights;
+    rightsGROUP.attributeId = this.attDef.id;
+    rightsGROUP.role = Role.GROUPADMIN;
+    rightsGROUP.rights = [];
+
+    if (this.readGroup) {
+      rightsGROUP.rights.push(ActionType.READ);
+    }
+
     if (this.writeGroup) {
-      list.push(this.generateRights(Role.GROUPADMIN, ActionType.WRITE));
+      rightsGROUP.rights.push(ActionType.WRITE);
     }
+
+    list.push(rightsGROUP);
+
+    const rightsFACILITY = {} as AttributeRights;
+    rightsFACILITY.attributeId = this.attDef.id;
+    rightsFACILITY.role = Role.FACILITYADMIN;
+    rightsFACILITY.rights = [];
+
+    if (this.readFacility) {
+      rightsFACILITY.rights.push(ActionType.READ);
+    }
+
     if (this.writeFacility) {
-      list.push(this.generateRights(Role.FACILITYADMIN, ActionType.WRITE));
+      rightsFACILITY.rights.push(ActionType.WRITE);
     }
 
+    list.push(rightsFACILITY);
+
+    // if (this.readSelf) {
+    //   list.push(this.generateRights(Role.SELF, ActionType.READ));
+    // }
+    // if (this.readSelfPublic) {
+    //   list.push(this.generateRights(Role.SELF, ActionType.READ_PUBLIC));
+    // }
+    // if (this.readSelfVo) {
+    //   list.push(this.generateRights(Role.SELF, ActionType.READ_VO));
+    // }
+    // if (this.readVo) {
+    //   list.push(this.generateRights(Role.VOADMIN, ActionType.READ));
+    // }
+    // if (this.readGroup) {
+    //   list.push(this.generateRights(Role.GROUPADMIN, ActionType.READ));
+    // }
+    // if (this.readFacility) {
+    //   list.push(this.generateRights(Role.FACILITYADMIN, ActionType.READ));
+    // }
+    //
+    // if (this.writeSelf) {
+    //   list.push(this.generateRights(Role.SELF, ActionType.WRITE));
+    // }
+    // if (this.writeSelfPublic) {
+    //   list.push(this.generateRights(Role.SELF, ActionType.WRITE_PUBLIC));
+    // }
+    // if (this.writeSelfVo) {
+    //   list.push(this.generateRights(Role.SELF, ActionType.WRITE_VO));
+    // }
+    // if (this.writeVo) {
+    //   list.push(this.generateRights(Role.VOADMIN, ActionType.WRITE));
+    // }
+    // if (this.writeGroup) {
+    //   list.push(this.generateRights(Role.GROUPADMIN, ActionType.WRITE));
+    // }
+    // if (this.writeFacility) {
+    //   list.push(this.generateRights(Role.FACILITYADMIN, ActionType.WRITE));
+    // }
```

Comment:
Is there some reason to keep these lines as comments?

Reply:
I was just trying to pump up my numbers. Just kidding, this is part of code that needed fixing and I replaced it with another, and forget it here.

Reply:
Removed